tests/rptest/services/admin.py

```diff
@@ -78,6 +97,189 @@ def ready(node):
     def _url(node, path):
         return f"http://{node.account.hostname}:9644/v1/{path}"
 
+    def get_redpanda_nodes(self) -> list[RedpandaNode]:
+        return [
+            RedpandaNode(ip=n.account.hostname, id=self.redpanda.idx(n))
+            for n in self.redpanda.nodes
```

Comment:
The small amount of typing saved by `RedpandaNode` doesn't seem worth it considering that it means there are now two different incompatible node types for the same thing. I think we should remove this (or discuss a different perhaps based on sub-classing ducktape node type). I don't think this needs to block the PR, but I was under the impression we'd remove it. Should we do this in a follow up PR?

Reply:
I gave up on this structure, replace input arguments with list[str] (host) and output with list[int] (node id)